elements/lisk-chain/src/data_access/storage.ts

```diff
@@ -341,28 +347,15 @@ export class Storage {
 		return accounts;
 	}
 
-	// TODO: Remove this with issue #5259
 	public async getDelegates(): Promise<AccountJSON[]> {
-		const stream = this._db.createReadStream({
-			gte: getFirstPrefix(DB_KEY_ACCOUNTS_ADDRESS),
-			lte: getLastPrefix(DB_KEY_ACCOUNTS_ADDRESS),
-		});
-		const accounts = await new Promise<AccountJSON[]>((resolve, reject) => {
-			const accountJSONs: AccountJSON[] = [];
-			stream
-				.on('data', ({ value }) => {
-					const { username } = value as AccountJSON;
-					if (username) {
-						accountJSONs.push(value);
-					}
-				})
-				.on('error', error => {
-					reject(error);
-				})
-				.on('end', () => {
-					resolve(accountJSONs);
-				});
-		});
+		// Data format for the registered delegates
```

Comment:
this `getDelegates` should be moved to `dpos` where it's used.
it has temporal logic 
```
// This list is before executing the current block in process
const originalDelegates = await this.chain.dataAccess.getDelegates();
```
this should be changed to below function, using state store instead.
then this `This list is before executing the current block in process` is no longer true, that it always get the consistent information in the state store

Reply:
But we don't have `db` access in the dpos, we access it through `chain` object. Don't get how can we move that function outside `lisk-chain`

Reply:
we have access to `stateStore.chain.get` and `stateStore.account.get`, so it doesn't need the direct access to `db` object